File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right


class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        
        def getGoodNodesCount(node, maxVal):
            if not node: 
                return 0
            
            res = 1 if node.val >= maxVal else 0
            maxVal = max(node.val, maxVal)
            
            left = getGoodNodesCount(node.left, maxVal)
            right = getGoodNodesCount(node.right, maxVal)
            
            return res + right + left

        return getGoodNodesCount(root, root.val)
        

File: Data Structures & Algorithms/balanced-binary-tree/submission-2.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {TreeNode} root
     * @return {boolean}
     */
    isBalanced(root) {
        let isOk = true;

        const getLevels = (root) => {
            if (!root) {
                return 0;
            }

            const left = getLevels(root.left);
            const right = getLevels(root.right);

            if (left - right > 1 || right - left > 1) {
                isOk = false;
            }

            return 1 + Math.max(left, right);
        };

        getLevels(root);

        return isOk;
    }
}
